### PR TITLE
chore: scaffolding incl ruff, mkdocs

### PR DESCRIPTION
**Pull request by blueraft**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -9,18 +9,93 @@ description = 'An example parser plugin for NOMAD.'
...
+dependencies = ["nomad-lab>=1.2.0"]
 
 [project.urls]
 homepage = "https://github.com/nomad-coe/nomad-parser-example-plugin"
 
 [project.optional-dependencies]
-tests = [
-    'pytest'
+dev = [
+    'ruff>=0.2.2',
```
Comment (ladinesa) [inline pyproject.toml:19]: should we make this the same as in nomad?

File: requirements.txt [hunk 2]
```diff
@@ -1,8 +0,0 @@
-# This is not an official pypi release. You need to add our gitlab package registry
```
Comment (ladinesa) [inline requirements.txt:1]: we do not need this file right?

File: pyproject.toml [hunk 3]
```diff
@@ -9,18 +9,93 @@ description = 'An example parser plugin for NOMAD.'
...
+    'mypy==1.0.1',
+    'pytest',
```
Comment (ladinesa) [inline pyproject.toml:21]: maybe we make versions consistent with nomad. The current setup requires it.

Comment (blueraft) [reply pyproject.toml:21]: I've relaxed the version requirements for this so it'll resolve with the current nomad setup.

File: requirements.txt (hunk 2 above)
Comment (blueraft) [reply requirements.txt:1]: yes, pip install directly from the toml file.

**Comment by blueraft** [comment]
@ladinesa can you merge this too? I don't have write permissions to do that.

**Pull request merged**